### ISSUE TITLE
Bug: [no-type-alias] template literals are unhandled

### ISSUE THREAD

**Issue by burtek** [typescript-eslint/typescript-eslint#5043]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/no-this-alias/

### Description

Currently when aliasing [template literal type](https://www.typescriptlang.org/docs/handbook/2/template-literal-types.html), an error is thrown with message "Type unhandled are not allowed"

![image](https://user-images.githubusercontent.com/2940958/169723681-071f4a56-fe99-4f5d-8c20-adeaf0856c7d.png)

This rule should recognise template literal type. Additionally, a new option `allowTemplateLiteral` should be introduced to control rule's behaviour similarly to other options in this rule.

### Fail

```typescript
// with allowTemplateLiteral: false (default)
type SlashCommand = `/${string}`
```

**Comment by JoshuaKGoldberg** [comment]
Posting back after reviewing #5092: I don't think there's a strong need for a new `allowTemplateLiteral` option. The rule already has basic handling for strings.

**Comment by burtek** [comment]
> The rule already has basic handling for strings

does it? Can you elaborate please?

**Comment by JoshuaKGoldberg** [comment]
I'll answer in #5095 - thanks for filing!
